deployment/helpers.go

```diff
@@ -177,12 +178,15 @@ func DeployContract[C any](
 		lggr.Errorw("Failed to deploy contract", "chain", chain.String(), "err", contractDeploy.Err)
 		return nil, contractDeploy.Err
 	}
-	_, err := chain.Confirm(contractDeploy.Tx)
-	if err != nil {
-		lggr.Errorw("Failed to confirm deployment", "chain", chain.String(), "Contract", contractDeploy.Tv.String(), "err", err)
-		return nil, err
+	var err error
+	if !chain.IsZk {
```

Comment:
what happens if zk?

Reply:
We _wait mined_ inside the wrapper because the transaction type is different. See https://github.com/smartcontractkit/chainlink-evm/blob/35c5921b859765c28422e9538a2546c98e87f8b4/gethwrappers/shared/generated/link_token/link_token_zksync.go#L47